src/ws/session.c

```diff
@@ -544,10 +544,9 @@ perform_tlscert (const char *rhost,
 
   debug ("start tls-cert authentication for cockpit-ws %u", getppid ());
 
-  if (strncmp (authorization, "tls-cert ", 9) == 0)
-    client_certificate = authorization + 9;
-  else
-    exit_init_problem (PAM_AUTH_ERR);
+  /* True, otherwise we wouldn't be here. */
+  assert (strncmp (authorization, "tls-cert ", 9));
+  client_certificate = authorization + 9;
```

Comment:
This is the file name on disk which contains the cert, not the cert itself, right? Might be worth renaming this to avoid confusion.

Reply:
It's also called "client-certificate" in the metadata blob...

Reply:
I changed the name.